backend/app/utils/jsonl_parser.py

```
@@ -0,0 +1,738 @@
+        
+        # 验证引用
+        for block in blocks:
+            ref = block.get_attribute("ref")
+            if ref and ref not in paragraph_map:
+                warnings.append(ParseWarning(
+                    message=f"Reference '{ref}' not found",
+                    line_number=block.line_number,
+                    context=f"Block type: {block.type}"
+                ))
+    
+    def parse_sync(self, content: str, options: Optional[ParseOptions] = None) -> ParseResult:
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)